packages/nx/src/config/misc-interfaces.ts

```diff
@@ -56,6 +56,7 @@ export type PackageJsonUpdates = {
     packages: {
       [packageName: string]: PackageJsonUpdateForPackage;
     };
+    confirmationPrompt?: string | boolean;
```

Comment:
It's nice that confirmationPrompts is assigned to a group of packages.

Over time, we're going to have multiple updates for updates that are related:

```
  1.0: {
    version: 1.0,
    packages: {
      react 18.0
    },
    confirmationPrompt: "Update to React 18?"
  },
  1.1: {
    version: 1.1,
    packages: {
      react: 18.1
    },
    confirmationPrompt: "Update to React 18.1?"
  }
```

When someone goes straight from 1.0 -> 1.2 they are going to see 2 prompts. When they're going from like 1.0 -> 2.4, they might very well see 4-5 prompts.

I think you should update `angular/migrations.json` going back to 15.1 (before Angular 15) to get a realistic idea of how this would look like to skip Angular 15 update. It's okay to even commit because this flag is hidden for now.

I expect there will be many prompts.
1. Update to Angular 15.0?
2. Update to `@angular-eslint/eslint-plugin` to 15.0? (Requires Angular 15.0)
3. Update to `jest-preset-angular` to 12.2.3? (Requires Angular 15.0)
4. Update ngrx to 15.0? (Requires Angular 15.0)
5. Update to Angular 15.1? (Requires Angular 15.0)

And there are wrong answers... There's basically only a few sane correct answers?
* Update all
* Update none
* Only update to 15.0, not 15.1

2-4 are basically attached to the answer of 1
5 is basically attached to the answer of 1-4

What does it look like if we have:

```
  1.0: {
    version: 1.0,
    packages: {
      angular/core: 15.0
    },
    confirmationPrompt: "Update to Angular 15?"
  },
  1.1: {
    version: 1.1,
    packages: {
      jest-preset-angular: 12.1
    },
    dependencies: [1.0]
  }
```

Reply:
Yeah, I have the Angular migrations changes in a separate branch for which I haven't created the PR yet. I agree some prompts are asked that shouldn't be asked at all, if Angular 15 was chosen, then all the related packages updates to support it should be updated, no questions asked. I think we discussed correlating groups to use the related answers already provided but I didn't remember the outcome of that and it's not in the design doc, so I didn't go for it.

I like the idea behind `dependencies`. I'll explore doing that.